registry/api/v2/urls.go

```diff
@@ -249,3 +272,160 @@ func appendValues(u string, values ...url.Values) string {
 
 	return appendValuesURL(up, values...).String()
 }
+
+// Following are states of forwarded header parser.
+// Here is a list of allowed transitions:
+//  *Element           -> Parameter
+//  *Parameter         -> KeyValueDelimiter
+//   KeyValueDelimiter -> Value
+//   Value             -> QuotedValue, PairEnd
+//   QuotedValue       -> EscapedCharacter, PairEnd
+//   EscapedCharacter  -> QuotedValue
+//  *PairEnd           -> Parameter, Element
+//
+// States marked with '*' are terminating. Any state could transition to a
+// a failure which isn't listed among states.
+const (
```

Comment:
Are these constants used outside of the parsing function? May just want to define them local to the function.